### ISSUE TITLE
Move popdict to dataframe

### ISSUE THREAD

**Issue by inc0** [InstituteforDiseaseModeling/covasim#7]
Right now pop dict is just that - dict. By moving this to dataftame we'll have access to better querying via pandas or full parallelization via dask later.

**Comment by gwincr11** [comment]
I started peeking at this out of curiosity. I am wondering if we can declare it as a dataframe on line 96 here:
https://github.com/InstituteforDiseaseModeling/covasim/blob/cfc73005f97acbefbf62437b30503aef36621abc/covasim/sim.py#L88-L102

I also saw pop_dict here, which seems to mutate the sim object? So we may need to do it there, or perhaps set up a `@property` setter so that it is always cast when being set? 
https://github.com/InstituteforDiseaseModeling/covasim/blob/master/covasim/people.py#L292

Could we use a Sciris dataframe or does it need to be Pandas? https://github.com/sciris/sciris/blob/develop/sciris/sc_dataframe.py

It seems like the Sciris stuff, which I believe is for working with the web app mainly is pretty tightly coupled to the simulator. I am wondering if a translation layer may allow for more flexibility as well, this is a very naive understanding currently.
